File: catkin_ws/src/Task2/src/constant.cpp
#include "ros/ros.h"
#include "std_msgs/Int32.h"

std_msgs::Int32 inc;
std_msgs::Int32 dec;
std_msgs::Int32 constant;


void inc_callback(std_msgs::Int32 message)
{
    inc.data = message.data;
}
void dec_callback(std_msgs::Int32 message)
{
    dec.data = message.data;
}

int main(int argc, char **argv)
{
    ros::init(argc, argv, "constant");
    ros::NodeHandle nh;

    ros::Subscriber sub_inc = nh.subscribe("inc", 10, inc_callback);
    ros::Subscriber sub_dec = nh.subscribe("dec", 10, dec_callback);
    ros::Publisher pub_const = nh.advertise<std_msgs::Int32>("const", 10);

    ros::Rate rate(1);
    while (ros::ok())
    {
        constant.data = inc.data + dec.data;
        pub_const.publish(constant);
        rate.sleep();
    }

    return 0;
}

File: catkin_ws/src/Task2/scripts/constant.py
#!/usr/bin/env python
import rospy
from std_msgs.msg import Int64
import message_filters

rospy.init_node("constant")

class constant:
    def __init__(self):
        self.constant = Int64()
        self.subs_inc = message_filters.Subscriber("/inc", Int64)
        self.subs_dec = message_filters.Subscriber("/dec", Int64)
        self.ts = message_filters.TimeSynchronizer([self.subs_inc, self.subs_dec], 1)
        self.ts.registerCallback(self.callback_func)
    def callback_func(self, inc, dec):
        self.constant = inc+dec

constant_obj = constant()
publisher = rospy.Publisher("/const", Int64, queue_size=1)
rate = rospy.Rate(1)

while not rospy.is_shutdown():
    publisher.publish(constant_obj.constant)
    rate.sleep()
